src/pynwb/__init__.py

```diff
@@ -251,7 +255,7 @@ def can_read(path: str):
              'doc': 'a path to a namespace, a TypeMap, or a list consisting paths to namespaces and TypeMaps',
              'default': None},
             {'name': 'file', 'type': [h5py.File, 'S3File'], 'doc': 'a pre-existing h5py.File object', 'default': None},
-            {'name': 'comm', 'type': "Intracomm", 'doc': 'the MPI communicator to use for parallel I/O',
+            {'name': 'comm', 'type': '~mpi4py.MPI.Intracomm', 'doc': 'the MPI communicator to use for parallel I/O',
```

Comment:
```suggestion
            {'name': 'comm', 'type': '~mpi4py.MPI.Intracomm', 'doc': 'the MPI communicator to use for parallel I/O',
```

I'm not sure `'~mpi4py.MPI.Intracomm'` will work correctly with the `docval` decorator. `docval` uses this to check type by checking for the classname, I believe the line in HDMF is here 

https://github.com/hdmf-dev/hdmf/blob/b4bdcef10071199ab28e6ad3822b7fd47bfab4b3/src/hdmf/utils.py#L99

and think '~mpi4py.MPI.Intracomm' may not match this check. You can try and pass an mpi Intracomm object to the function and see if docval raises an error or not. When possible we list the actual class in the `type` key for docval but in this case we are just specifying the name of the class to avoid having to import mpi4py just to use it in docval.

Reply:
I don't think we are currently covering mpi in our unit tests, so this probably does not get picked up by tests yet. We have short tutorial for mpi, but that one does not actually run anything. 

https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/gallery/advanced_io/parallelio.py 

If I remember correctly, this is because testing with mpi requires a custom install for h5py to build against HDF5 with mpi enabled.

Reply:
I agree. I think this will not work unfortunately. We could update docval to allow us to specify a classname with module prefix. This would require some testing, but it could be useful in allowing us to be more explicit about which class we mean to allow. Or we could hardcode an exception in docval for interpreting "Intracomm" as ":py:class~`mpi4py.MPI.Intracomm`" when creating the sphinx docstring. That's easier. Either approach is fine with me.

Reply:
I would just leave it as `'Intracomm'` for now here. We are using the same logic in a couple of places, e.g., just two lines above for `'S3File'`. If this is a problem for the Sphinx link checker, then we could maybe just make it a list of types  `['Intracomm', ]` to work around sphinx checking the link. 

> it could be useful in allowing us to be more explicit about which class we mean to allow

I agree that would be a nice-to-have feature, but I don't think it is a priority. 

> we could hardcode an exception in docval

That seems brittle and would require exception for other types too.

Reply:
I see, that makes sense. Yes I can keep it as `'Intracomm'` for now and add an exception/workaround for the link checker.

Reply:
Even having a list of types `['Intracomm', ]` didn't seem to fix the reference warnings. I'm not sure if the `S3File` worked because there was another valid reference in the list? I added it to the `nitpick_ignore` section of the `conf.py` file as a potential solution 

I also moved the `nitpicky=True` flag to `conf.py` so it's all in the same place and consistent across make/workflow files.

Reply:
Thanks, that solution looks reasonable to me.